src/shared_modules/utils/windowsHelper.h

```diff
@@ -136,6 +169,59 @@ namespace Utils
         return ret;
     }
 
+
+    /* Reference: https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_2.6.0.pdf */
+    static std::string parseRawSmbios(const BYTE* rawData, const DWORD rawDataSize)
+    {
+        std::string serialNumber;
+        DWORD offset{0};
+
+        while (offset < rawDataSize && serialNumber.empty())
+        {
+            if (offset + sizeof(SMBIOSStructureHeader) >= rawDataSize)
+            {
+                break;
+            }
+
+            SMBIOSStructureHeader header{};
+            memcpy(&header, rawData + offset, sizeof(SMBIOSStructureHeader));
```

Comment:
consider this check to avoid problems with an older version of dmtf with fewer fields.

```
if (header.FormattedAreaLength + offset >= rawDataSize ||                                                                                                                                 
    sizeof(SMBIOSBaseboardInfoStructure) + offset >= rawDataSize)                                                                                                                         
{                                                                                                                                                                                         
    break;                                                                                                                                                                                
}
```